clifford/jit_func.py

```diff
@@ -0,0 +1,64 @@
+
+import ast
+import astpretty
+import inspect
+from ._numba_utils import njit
+from ._ast_transformer import GATransformer
+
+
+class jit_func(object):
+    """
+    This is a JIT decorator that re-writes the AST and then numba JITs
+    the resulting function.
+    """
+    def __init__(self, layout, ast_debug=False):
+        self.layout = layout
+        self.ast_debug = ast_debug
+
+    def __call__(self, func):
+        # Get the function source
+        fname = func.__name__
+        source = inspect.getsource(func)
+        # Remove the decorator first line.
+        source = '\n'.join(source.splitlines()[1:])
+        # Remove the indentation
+        indentation = source.splitlines()[0].find('def')
+        print(indentation)
+        source = '\n'.join([line[indentation:] for line in source.splitlines()])
+
+        # Re-write the ast
+        tree = ast.parse(source)
+        if self.ast_debug:
+            print('\n\n\n\n TRANSFORMING FROM \n\n\n\n')
+            astpretty.pprint(tree)
+
+        tree = GATransformer().visit(tree)
+        ast.fix_missing_locations(tree)
+
+        if self.ast_debug:
+            print('\n\n\n\n TRANSFORMING TO \n\n\n\n')
+            astpretty.pprint(tree)
+
+        # Set things up into locals and globals so that they JIT ok...
+        locals_dict = {'as_ga': self.layout.as_ga_value_vector_func,
+                       'ga_add': self.layout.overload_add_func,
+                       'ga_sub': self.layout.overload_sub_func,
+                       'ga_mul': self.layout.overload_mul_func,
+                       'ga_xor': self.layout.overload_xor_func,
+                       'ga_or': self.layout.overload_or_func}
+        globs = globals()
+        for k, v in locals_dict.items():
+            globs[k] = v
```

Comment:
This updates the global globals, which is unlikely what you want to do.

This might work:

```suggestion
        globs = collections.ChainMap(locals_dict, globals())
```